src/javascripts/modules/pills.js

```diff
@@ -6,28 +6,43 @@ export default class Pills {
   }
 
   setVars() {
-    this.ticking = false
-    this.lastScrollY = 0
+    this.firstIntersection = true
+    this.titleBox = document.querySelector('.title-box')
+    this.greenTablet = document.querySelector('.green-tablet')
   }
 
   setupListeners() {
     window.addEventListener('load', () => this.handleLoad())
-    window.addEventListener('scroll', () => this.handleScroll())
+    this.createObserver()
   }
 
   handleLoad() {
     this.timeout = setTimeout(() => this.animatePills(), 500)
     this.releaseSqueezeTimeout = setTimeout(() => this.releasePillSqueeze(), 3500)
   }
 
-  handleScroll() {
-    this.lastScrollY = window.scrollY
-    if (!this.ticking) {
-      window.requestAnimationFrame(() => {
-        this.shiftPills()
-      })
+  createObserver() {
+    this.options = {
+      root: null,
+      rootMargin: "20px",
+      threshold: 0,
     }
-    this.ticking = true
+
+    this.observer = new IntersectionObserver(this.handleIntersect.bind(this), this.options)
+    this.observer.observe(this.greenTablet)
+  }
+
+  handleIntersect(entries, observer) {
+    entries.forEach((entry) => {
+      if (this.firstIntersection) {
+        //preventing pill shift animation onLoad
+      } else if (entry.intersectionRatio > 0 && !this.titleBox.classList.contains('pills-move-in')) {
+          this.titleBox.classList.add('pills-move-in')
+      } else if (entry.intersectionRatio > 0 && this.titleBox.classList.contains('pills-move-in')) {
```

Comment:
Can these `else if` blocks now be `&& !this.firstIntersection` instead of checking for the classname?